### ISSUE TITLE
[BUG] Amount in ERC-20 transfers interpreted as CSPR currency

### ISSUE THREAD

**Issue by andrzej-casper** [make-software/casper-wallet#743]
### TL;DR

For contract's argument named `amount` value is always interpreted as CSRP - this is really confusing for ERC-20 calls.

### Problem description

I got asked why calling ERC-20's transfer is displaying wrong amount:

![image](https://github.com/make-software/casper-wallet/assets/121791569/0d83b06b-0539-4609-8dc2-2340d0868cf1)

So I [located that part in UI](https://github.com/make-software/casper-wallet/blob/v1.4.1/src/apps/signature-request/pages/sign-deploy/sign-deploy-content.tsx#L181-L186), and it appears `DeployValue` is using `isKeyOfCurrencyValue()` to determine if CSPR price is used:

![image](https://github.com/make-software/casper-wallet/assets/121791569/d678fa32-1d34-4f29-8aba-ce191ea4c444)

![image](https://github.com/make-software/casper-wallet/assets/121791569/28176dbe-76c0-4cdd-a733-f7a28b913837)

I think argument should not be interpreted basing solely on its name.

### Required information

 - Casper Wallet Version: 1.4.1
